hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/NearCachedClientCacheProxy.java

```diff
@@ -833,35 +823,37 @@ public void handle(ClientMessage clientMessage) {
             extends CacheAddNearCacheInvalidationListenerCodec.AbstractEventHandler
             implements EventHandler<ClientMessage> {
 
-        private final RepairingHandler repairingHandler;
-
-        public RepairableNearCacheEventHandler() {
-            getRepairingTask().deregisterHandler(nameWithPrefix);
-            this.repairingHandler = getRepairingTask().registerAndGetHandler(nameWithPrefix, nearCache);
-        }
+        private volatile RepairingHandler repairingHandler;
 
         @Override
         public void beforeListenerRegister() {
-
+            if (supportsRepairableNearCache()) {
+                RepairingTask repairingTask = getContext().getRepairingTask(CacheService.SERVICE_NAME);
+                repairingHandler = repairingTask.registerAndGetHandler(nameWithPrefix, nearCache);
+            } else {
+                RepairingTask repairingTask = getContext().getRepairingTask(MapService.SERVICE_NAME);
+                repairingTask.deregisterHandler(nameWithPrefix);
+            }
         }
 
         @Override
         public void onListenerRegister() {
+            // NOP
         }
 
         @Override
         public void handle(String name, Data key, String sourceUuid, UUID partitionUuid, long sequence) {
+            assert repairingHandler != null;
```

Comment:
Those assertions will create a code coverage branch on the class level, which needs to be tested. Since we'll just get an `AssertionError` instead of a `NPE`, I don't think this `assert` will have much benefit. It's not that there is a silent error will be prevented by it. Same in the other `handle()` method.

Reply:
fixed